docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+- `Chore`: This class manages the data of data type chore in memory.
+- `Recipe`: This class manages the data of data type recipe in memory.
+- `Storage`: This class reads data from and writes data back into a text file for future uses.
 
-## User Stories
+### 3.2. Ui Component
+![Ui Component](images/UI_Component.png)
+
+API: `Ui.java`
+ 
+The `Ui` component is a singleton class where all interaction will be made through this component
+ 
+The `Ui` component,
+
+* Executes user commands using the command component
+* Listens for changes and outputs messages from the Command component
+
+### 3.3. Logic Component
+
+![Logic Component](images/logic.png)
+
+1. `Kitchen Helper` uses `Parser` class to parse the user command.
+2. This results in a command object return back which is executed by `Kitchen Helper`.
+3. The command execution can affect the object (e.g. adding an ingredient).
+4. The result of the command execution is encapsulated as a `CommandResult` object which is passed back to `Ui` to display the message.
+
+### 3.4. Model Component
+
+![Model Component](images/model.png)
+* Stores the ingredient data.
+* Stores the recipe data.
+* Stores the chore data.
+
+### 3.5. Storage Component
+
+![Storage Class Diagram](images/Storage.png)
+
+A Storage object is created by the KitchenHelper class to handle the loading and saving of ingredients, recipes and chores data.
+
+The Storage() method acts as a constructor with filepaths to local save files for ingredients, recipes and chores data.
+
+The getIngredientData(), getRecipeData() and getChoreData() methods are used to read saved data from local files into the current session of KitchenHelper. loadingIngredients() and loadingRecipeItems() methods are called in getIngredientData() and getRecipeData() respectively to sort out which Ingredient object class each object belongs to.
+
+The saveIngredientData(), saveRecipeData() and saveChoreData() methods write the current state of KitchenHelper into the local save files by calling them in command classes such as AddChoreCommand and DeleteIngredientCommand.
+
+### 3.6. Common Classes 
+Classes used by multiple components are in the `seedu.kitchenhelper.object` package.
+
+## 4. Implementation
+This section describes some details on how the features are being implemented. All recipe/ ingredient/ chore-related features can be broken down into 4 distinct functionality, addition, listing, deletion and searching.
+
+### 4.1.Ingredient-related Features
+#### 4.1.1. Addition of ingredient
+
+The addition of the ingredient feature allows the user to keep track of the ingredients in the ingredient’s list.   
+For example, `addingredient /n beef /c meat /q 2 /p 20 /e 18/02/2020` will add the ingredient `beef` 
+which have the following attributes:  category `meat`, quantity `2`, price `$20` and expiry `18/02/2020`  
+
+##### Implementation
+
+![addIngredientCommand](images/addIngredientCommand.png) 
```

Comment:
Maybe you can use an "X" to indicate the point at which the objects ceases to be used? Because at any time only 1 command is "alive/being used"..  😄